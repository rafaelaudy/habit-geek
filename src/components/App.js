import React from "react";
import { Router } from "@reach/router";

import { getCurrentWeek } from "@rafael.audy/habit-geek-utils/utils/dateUtils";

import UserContainer from "./user/UserContainer";
import ForgotPassword from "./user/ForgotPassword";
import SignUpContainer from "./user/SignUpContainer";
import DashboardContainer from "./dashboard/DashboardContainer";
import Layout from "./layout/Layout";
import HistoryContainer from "./history/HistoryContainer";

import "./App.scss";

function App({ currentWeek, startNewWeek, updateCurrentWeekStatuses }) {
  if (currentWeek !== getCurrentWeek()) {
    startNewWeek();
  } else {
    updateCurrentWeekStatuses();
  }

  return (
    <div>
      <Router>
        <Layout path="/habit-geek" default>
          <DashboardContainer
            path="/habit-geek/habits"
            default
          ></DashboardContainer>
          <UserContainer path="/habit-geek/profile"></UserContainer>
          <UserContainer path="/habit-geek/login"></UserContainer>
          <ForgotPassword path="/habit-geek/forgotpassword"></ForgotPassword>
          <SignUpContainer path="/habit-geek/signup"></SignUpContainer>
          <HistoryContainer path="/habit-geek/history"></HistoryContainer>
        </Layout>
      </Router>
    </div>
  );
}

export default App;
