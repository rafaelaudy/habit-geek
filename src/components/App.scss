@import "bootstrap/scss/bootstrap";

.mobile-size-container {
  max-width: 340px;
  padding: 15px;
  margin: auto;
}

.large-size-container {
  max-width: 650px;
  padding: 15px;
  margin: auto;
}

.mobile-size-container,
.large-size-container {
  h2 {
    text-align: center;
  }
}

.was-validated .form-control:valid,
.was-validated .form-control:invalid {
  background-image: none;
}

.icon-button {
  width: 44px;
  padding-bottom: 10px;
}
