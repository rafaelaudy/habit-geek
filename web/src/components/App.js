import React from "react";
import { Router } from "@reach/router";

import { getCurrentWeek } from "@habit-geek/shared/utils/dateUtils";

import ProfileContainer from "./user/ProfileContainer";
import DashboardContainer from "./dashboard/DashboardContainer";
import Layout from "./layout/Layout";
import HistoryContainer from "./history/HistoryContainer";

import "./App.scss";

function App({ currentWeek, startNewWeek }) {
  if (currentWeek !== getCurrentWeek()) {
    startNewWeek();
  }

  return (
    <div>
      <Router>
        <Layout path="/habit-geek" default>
          <DashboardContainer
            path="/habit-geek/habits"
            default
          ></DashboardContainer>
          <ProfileContainer path="/habit-geek/profile"></ProfileContainer>
          <HistoryContainer path="/habit-geek/history"></HistoryContainer>
        </Layout>
      </Router>
    </div>
  );
}

export default App;
