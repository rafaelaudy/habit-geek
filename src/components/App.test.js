import React from "react";
import { shallow } from "enzyme";
import App from "./App";

jest.mock("@rafael.audy/habit-geek-utils/utils/dateUtils", () => ({
  getCurrentWeek: jest.fn().mockReturnValue("y1w1")
}));

describe("App component", () => {
  it("Renders static elements", () => {
    const app = shallow(<App currentWeek="y1w1" updateCurrentWeekStatuses={()=>{}}/>);
    expect(app).toMatchSnapshot();
  });

  it("starts a new week when the current date is different then what was stored", () => {
    const startNewWeekMock = jest.fn();
    shallow(<App currentWeek="y1w0" startNewWeek={startNewWeekMock} />);
    expect(startNewWeekMock).toHaveBeenCalled();
  });

  it("updates current week statuses", () => {
    const updateCurrentWeekStatusesMock = jest.fn();
    shallow(<App currentWeek="y1w1" updateCurrentWeekStatuses={updateCurrentWeekStatusesMock} />);
    expect(updateCurrentWeekStatusesMock).toHaveBeenCalled();
  });
});
