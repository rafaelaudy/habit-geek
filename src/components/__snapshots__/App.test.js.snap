// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders static elements 1`] = `
<div>
  <Router>
    <Layout
      default={true}
      path="/habit-geek"
    >
      <Connect(Dashboard)
        default={true}
        path="/habit-geek/habits"
      />
      <User
        path="/habit-geek/profile"
      />
      <User
        path="/habit-geek/login"
      />
      <Connect(History)
        path="/habit-geek/history"
      />
    </Layout>
  </Router>
</div>
`;
